AmplifyTools/AmplifyXcode/Sources/AmplifyXcodeCore/Support/XcodeProject.swift

```diff
@@ -48,22 +62,41 @@ struct XcodeProject {
     func synchronize() throws {
         try project.pbxproj.write(path: Path(pbxProjFilePath), override: true)
     }
+
+    private func resolveTarget(_ target: XcodeProjectTarget) throws -> PBXTarget {
+        let targetName: String
+        switch target {
+        case .primary:
+            targetName = project.mainProject()?.name ?? "primary"
+        case .named(let name):
+            targetName = name
+        }
+
+        if let targetRef = project.targets(named: targetName, ofType: .application).first {
+            return targetRef
+        }
+
+        throw XcodeProjectError.targetNotFound(name: targetName)
```

Comment:
Instead of throwing an error, since this is an expected case this function and others could return an enum for the status and it would support the 2 paths for this command to either add files to the Xcode project or to ask the developer to do it manually. I do not think it is an error as much as one of these 2 paths.